### PR TITLE
added user signup api

### PR DESCRIPTION
**Pull request by rkpatra201**
Task: https://github.com/learning-project-01/bookstore-app/issues/10

image

```bash
curl --location 'localhost:8080/user/signup' \
--header 'Content-Type: application/json' \
--data-raw '{
    "email":"abcd@email.com",
    "password": "abc@123",
    "firstName":"John",
    "lastName": "Doe",
    "role": "CUSTOMER" 
}'
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -56,6 +56,10 @@
 			<artifactId>flyway-core</artifactId>
 			<version>${flyway.version}</version>
 		</dependency>
+		<dependency>
```
Comment (rkpatra201) [inline pom.xml:59]: for brcrypt password encoding

File: src/main/java/com/example/bookstoreapp/services/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  @Autowired
+  private UserEntityRepository userEntityRepository;
+
+  private PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+
+  @Override
+  public User createUser(User user) {
+    user.setId(System.nanoTime());
+    String rawPassword = user.getPassword();
+    user.setPassword(passwordEncoder.encode(rawPassword));
```
Comment (rkpatra201) [inline src/main/java/com/example/bookstoreapp/services/impl/UserServiceImpl.java:24]: bcrypt hash password

**Review by rkpatra201 — COMMENTED** [review]
added comments

**Comment by Subhransupanda2000** [comment]
I saw your codes . But i have created all api individually .ex-your all controller layer are in same package called controller but i have all controller layers in different packages and for each api i have created individual flyway and db/scripts .should i follow your approach or no need of correction

**Comment by rkpatra201** [comment]
> I saw your codes . But i have created all api individually .ex-your all controller layer are in same package called controller but i have all controller layers in different packages and for each api i have created individual flyway and db/scripts .should i follow your approach or no need of correction

Yes follow same approach as mine

**Comment by Subhransupanda2000** [comment]
I didnt get your createUser method of userImpl class

**Pull request merged**